tests/api/test_checkout.py

```diff
@@ -434,6 +434,93 @@ def test_checkout_available_shipping_methods(
     assert data["availableShippingMethods"] == [{"name": shipping_method.name}]
 
 
+def test_checkout_available_shipping_methods_with_net_price_displayed(
+    monkeypatch, api_client, checkout_with_item, address, shipping_zone, site_settings
+):
+    shipping_method = shipping_zone.shipping_methods.first()
+
+    apply_taxes_to_shipping_mock = mock.Mock()
+    apply_taxes_to_shipping_mock.return_value = TaxedMoney(
+        net=shipping_method.price, gross=Money(13, "USD")
+    )
+
+    manager_mock = mock.MagicMock(apply_taxes_to_shipping=apply_taxes_to_shipping_mock)
+    monkeypatch.setattr(
+        "saleor.graphql.checkout.types.get_extensions_manager",
+        lambda **kwargs: manager_mock,
+    )
+    site_settings.display_gross_prices = False
+    site_settings.save()
+    checkout_with_item.shipping_address = address
+    checkout_with_item.save()
+
+    query = """
+    query getCheckout($token: UUID!) {
+        checkout(token: $token) {
+            availableShippingMethods {
+                name
+                price {
+                    amount
+                }
+            }
+        }
+    }
+    """
+    variables = {"token": checkout_with_item.token}
+    response = api_client.post_graphql(query, variables)
+    content = get_graphql_content(response)
+    data = content["data"]["checkout"]
+
+    assert manager_mock.apply_taxes_to_shipping.assert_called_once
+    assert data["availableShippingMethods"] == [
+        {"name": "DHL", "price": {"amount": shipping_method.price.amount}}
+    ]
+
+
+def test_checkout_available_shipping_methods_with_gross_price_displayed(
```

Comment:
Let's try to use `pytest.mark.parametrize` and create one test for both of these cases